staging/src/k8s.io/apiextensions-apiserver/examples/client-go/pkg/client/informers/externalversions/factory.go

```diff
@@ -47,20 +50,27 @@ type sharedInformerFactory struct {
 
 // NewSharedInformerFactory constructs a new instance of sharedInformerFactory
 func NewSharedInformerFactory(client versioned.Interface, defaultResync time.Duration) SharedInformerFactory {
-	return NewFilteredSharedInformerFactory(client, defaultResync, v1.NamespaceAll, nil)
+	return NewFilteredSharedInformerFactory(client, defaultResync, v1.NamespaceAll, nil, nil)
 }
 
 // NewFilteredSharedInformerFactory constructs a new instance of sharedInformerFactory.
 // Listers obtained via this SharedInformerFactory will be subject to the same filters
 // as specified here.
-func NewFilteredSharedInformerFactory(client versioned.Interface, defaultResync time.Duration, namespace string, tweakListOptions internalinterfaces.TweakListOptionsFunc) SharedInformerFactory {
+func NewFilteredSharedInformerFactory(client versioned.Interface, defaultResync time.Duration, namespace string, tweakListOptions internalinterfaces.TweakListOptionsFunc,
```

Comment:
why does `NewFilteredSharedInformerFactory` get the `resyncConfig` param, but `NewSharedInformerFactory` doesn't?

Instead of adding more args, we should switch to another extension pattern, maybe `informer.WithCustomResyncs`.

Reply:
Adding the parameter to `NewSharedInformerFactory` would cause a huge refactor accross the Kubernetes code-base, because all consumers of this factory almost exclusivley use this method to instantiate a new factory. 

Changing the arguments of `NewFilteredSharedInformerFactory` does not break any existing functionality as it is not used verbatim. So this seemed like a natural fit as it also already incorporates custom list options.

Reply:
What I wanted to say is that adding another param is the wrong pattern.

Reply:
Ok, so you would suggest adding a seperate method e.g. `NewSharedInformerFactoryWithCustomResync`instead?

Reply:
No. Rather I would prefer some functional style pattern:

`NewSharedInformerFactory(...).WithTweakListOptions(...).WithCustomResyncConfig(...)`.